src/mono/mono/mini/aot-compiler.c

```diff
@@ -14198,121 +14196,6 @@ mono_read_method_cache (MonoAotCompile *acfg)
 	return;
 }
 
-typedef struct {
-	GHashTable *cache;
-	GHashTable *stats;
-	gboolean emit_inflated_methods;
-	MonoAssembly *inflated_assembly;
-} MonoAotState;
-
-static MonoAotState *
-alloc_aot_state (void)
-{
-	MonoAotState *state = g_malloc (sizeof (MonoAotState));
-	// FIXME: Should this own the memory?
-	state->cache = g_hash_table_new (g_str_hash, g_str_equal);
-	state->stats = g_hash_table_new (g_str_hash, g_str_equal);
-	// Start in "collect mode"
-	state->emit_inflated_methods = FALSE;
-	state->inflated_assembly = NULL;
-	return state;
-}
-
-static void
-free_aot_state (MonoAotState *astate)
-{
-	g_hash_table_destroy (astate->cache);
-	g_free (astate);
-}
-
-static void
-mono_add_deferred_extra_methods (MonoAotCompile *acfg, MonoAotState *astate)
-{
-	GHashTableIter iter;
-	gchar *name = NULL;
-	MonoMethod *method = NULL;
-
-	acfg->dedup_emit_mode = TRUE;
-
-	g_hash_table_iter_init (&iter, astate->cache);
-	while (g_hash_table_iter_next (&iter, (gpointer *) &name, (gpointer *) &method)) {
-		add_method_full (acfg, method, TRUE, 0);
-	}
-	return;
-}
-
-static void
-mono_setup_dedup_state (MonoAotCompile *acfg, MonoAotState **global_aot_state, MonoAssembly *ass, MonoAotState **astate, gboolean *is_dedup_dummy)
-{
-	if (!acfg->aot_opts.dedup_include && !acfg->aot_opts.dedup)
-		return;
-
-	if (global_aot_state && *global_aot_state && acfg->aot_opts.dedup_include) {
-	// Thread the state through when making the inflate pass
-		*astate = *global_aot_state;
-	}
-
-	if (!*astate) {
-		*astate = alloc_aot_state ();
-		*global_aot_state = *astate;
-	}
-
-	acfg->dedup_cache = (*astate)->cache;
-	acfg->dedup_stats = (*astate)->stats;
```

Comment:
`mono_dedup_log_stats` needs `MonoAotCompile:dedup_stats` to be non-NULL and nothing else initializes it.

Reply:
Removed that as well.